domain/application/service/application_test.go

```diff
@@ -1433,6 +1433,47 @@ func (s *applicationServiceSuite) TestSetWorkloadUnitStatusInvalidStatus(c *gc.C
 	c.Assert(err, gc.ErrorMatches, `.*unknown workload status "allocating"`)
 }
 
+func (s *applicationServiceSuite) TestGetUnitWorkloadStatusesForApplication(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	now := time.Now()
+
+	appUUID := applicationtesting.GenApplicationUUID(c)
+	s.state.EXPECT().GetUnitWorkloadStatusesForApplication(gomock.Any(), appUUID).Return(
```

Comment:
You could probably make this test simpler by just saving the data as a var and checking its the same on both sides.

Reply:
Except they aren't quite equal, different types